### PR TITLE
Adding bulk-user-export-tool source code

### PR DESCRIPTION
**Pull request by mevan-karu**

### CODE REVIEW

File: bulk-user-export-tool/scim-bulk-user-export-tool/pom.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (emswbandara) [inline bulk-user-export-tool/scim-bulk-user-export-tool/pom.xml:1]: add license?

File: bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+package org.wso2.scim.bulk.export;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.*;
```
Comment (emswbandara) [inline bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java:28]: remove * import

File: bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    private static final String SCIM_USER_ENDPOINT = "scim2/Users";
+    private static final String PATH_SEPARATOR = "/";
+    private static final String NONE = "none";
+    private static final String DEFAULT_CSV_FILE = "users.csv";
+
+    public static void main(String args[]) throws IOException, KeyStoreException,
+            NoSuchAlgorithmException, KeyManagementException, URISyntaxException {
+
+        if (args.length < 6) {
+            System.out.println("Invalid arguments! Please provide valid arguments to host address, username and password.");
```
Comment (emswbandara) [inline bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java:69]: remove souts and change this to a log

File: bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                        }
+                        });
+            }
+            CsvSchema csvSchema = csvSchemaBuilder.build().withHeader();
+
+            CsvMapper csvMapper = new CsvMapper();
+            csvMapper.writerFor(ArrayNode.class)
+                    .with(csvSchema)
+                    .writeValue(new File(csvDirectory), usersArrayNode);
+            System.out.println("User information was successfully written to : " + csvDirectory + " file.");
```
Comment (emswbandara) [inline bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java:145]: remove souts and change to logs

File: bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        } else {
+            System.out.println(request.getMethod() + " Request to "+ request.getURI().toString() +
```
Comment (emswbandara) [inline bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java:147]: remove souts and change to logs

File: bulk-user-export-tool/scim-bulk-user-export-tool/pom.xml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.wso2.scim.bulk.export</groupId>
+    <artifactId>scim-bulk-user-export-tool</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (omindu) [inline bulk-user-export-tool/scim-bulk-user-export-tool/pom.xml:9]: 1.0.0-SNAPSHOT

File: bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+            builder.setParameter("attributes", attributes);
+        }
+
+        HttpGet request = HttpClient.createGetHttpRequest(builder, username, password);
+
+        final SSLContext sslContext = new SSLContextBuilder()
+                .loadTrustMaterial(null, (x509CertChain, authType) -> true)
+                .build();
+
+        CloseableHttpClient client = HttpClientBuilder.create()
```
Comment (omindu) [inline bulk-user-export-tool/scim-bulk-user-export-tool/src/main/java/org/wso2/scim/bulk/export/BulkExportUsers.java:105]: Use try with resource to close the client

File: bulk-user-export-tool/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# Bulk User Export Tool
+
+Java client that can export a bulk of users as a comma-separated values (.csv) using SCIM 2.0 APIs.
+
+## Table of contents
+
+- [Download and build](#download-and-build)
+- [Exporting user data from WSO2 IS](#exporting-user-data-from-wso2-is)
```
Comment (ayshsandu) [inline bulk-user-export-tool/README.md:8]: Let's reorder the content.
Bring "Exporting user data from WSO2 IS" section to the top. A user should be able to use the binary to consume the tool, without needing to build it. If someone needs to build the tool(optionally with modification) then can give the steps how to do that.

File: bulk-user-export-tool/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+directory.
+Application distributions are named as `scim-bulk-user-export-tool-LATEST-jar-with-dependencies.jar`.
+
+## Exporting user data from WSO2 IS
+
+Follow these steps to export user data from WSO2 IS
+
+1. Start the WSO2 IS. (Note : This should SCIM 2.0 supported and enabled IS version(5.3.0 or higher))
+2. Build the tool from source by following steps in [Building from source](#building-from-source).
+3. Copy `scim-bulk-user-export-tool-LATEST-jar-with-dependencies.jar` inside `target` directory to scripts
```
Comment (ayshsandu) [inline bulk-user-export-tool/README.md:34]: At this step, point to the downloadable binary, and optionally mention if user build from the source then use the .jar in the target.

File: bulk-user-export-tool/scim-bulk-user-export-tool/scripts/start.sh [hunk 10]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+read -p "Enter host address of WSO2 IS (Required) (ex: https://wso2.is/9443) : " host
```
Comment (ayshsandu) [inline bulk-user-export-tool/scim-bulk-user-export-tool/scripts/start.sh:3]: https://wso2.is/9443 --> https://localhost:9443, the familier format

**Pull request merged**